spec/support/shared/examples/msf/core/exploit/sqli/sqli_time_based_spec.rb

```diff
@@ -0,0 +1,22 @@
+RSpec.shared_examples 'TimeBasedBlind' do |sqli_class|
+  let(:timebased_class) do
+    sqli_class
+  end
+  let(:datastore) do
+    { 'SqliDelay' => 1.0 }
+  end
+  let(:sqli_obj) do
+    creation_time = Time.now
+    timebased_class.new(datastore, {}, {}) do |payload|
+      delay = payload[/\d+(?:.?\d*)?/].to_f
+      allow(Time).to receive(:now).and_return(creation_time + delay)
+    end
```

Comment:
Out of interest, was Timecop considered? It provides the ability to freeze time and time travel. There's existing examples within the codebase and their readme https://github.com/travisjeffery/timecop